File: 01_Language/01_Functions/python/call_user_func.py
# coding: utf-8

def call_user_func(cb, *args):
    return cb(*args)


if __name__ == '__main__':
    def barber(t):
        print('You wanted a {} haircut, no problem'.format(t))


    call_user_func(barber, "mushroom")
    call_user_func(barber, "shave")
    print(call_user_func(lambda x, y: x + y, 100, 200))

File: 01_Language/01_Functions/js/call_user_func.js
function call_user_func(cb) {
    let args = Array.prototype.slice.call(arguments, 1)
    return cb(...args)
}

function barber(type) {
    console.log(`You wanted a ${type} haircut, no problem`);
}

call_user_func(barber, "mushroom");
call_user_func(barber, "shave");
